gateway/gateway.proto

```diff
@@ -127,10 +133,37 @@ message EvaluateRequest {
 // EvaluateResponse returns the result of evaluating a transaction.
 message EvaluateResponse {
     // The response that is returned by the transaction function, as defined
-    // in peer/proposal_response.proto
+    // in peer/proposal_response.proto.
     protos.Response result = 1;
 }
 
+// SignedChaincodeEventsRequest contains a serialized ChaincodeEventsRequest message, and a digital signature for the
+// serialized request message.
+message SignedChaincodeEventsRequest {
+    // Serialized ChaincodeEventsRequest message.
+    bytes request = 1;
+    // Signature for request message.
+    bytes signature = 2;
+}
+
+// ChaincodeEventsRequest contains details of the chaincode events that the caller wants to receive.
+message ChaincodeEventsRequest {
+    // Identifier of the channel this request is bound for.
+    string channel_id = 1;
+    // Name of the chaincode for which events are requested.
+    string chaincode_id = 2;
+    // Client requestor identity.
+    bytes identity = 3;
+}
+
+// ChaincodeEventsResponse returns a chaincode event.
+message ChaincodeEventsResponse {
+    // Chaincode event emitted by the requested chaincode.
+    protos.ChaincodeEvent event = 1;
+    // Block number that contains the transaction that emitted this chaincode event.
+    uint64 block_number = 2;
```

Comment:
I am not certain about the motivation behind including the `block_number` here but one use I can think of is - when the client wants to reconnect after some failure, it would know till which block number it had consumed the events and from where to start. If that's the intent, I believe that including all the events in the block in a single response makes it easy. Not sure how it is exposed by the current SDKs and will it break the backward compatibility?

Reply:
Block number is there to allow resuming listening from a specific block position. This capability is covered by [FABGW-22](https://jira.hyperledger.org/browse/FABGW-22). For this story, it supports logging/debugging from the client end.

Reply:
Including all the events for a given block in a single message is not the way it is exposed to the client application, but it does make it easier for the client SDK to deal with so I'll change it to behave that way at the message level.